sdks/go/pkg/beam/core/runtime/metricsx/metricsx_test.go

```diff
@@ -0,0 +1,165 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metricsx
+
+import (
+	"testing"
+	"time"
+
+	"github.com/apache/beam/sdks/go/pkg/beam/core/metrics"
+	pipepb "github.com/apache/beam/sdks/go/pkg/beam/model/pipeline_v1"
+)
+
+func TestCounterExtraction(t *testing.T) {
```

Comment:
Test names are related to the code as much as possible. In this case, the thing we're properly testing is FromMonitoringInfos, and specifically Counters so `func TestFromMonitoringInfos_Counters` is an idiomatic test name.

There's also an opportunity for subtests... but that's difficult here without generics right now., and that's not until 2022. You can see more here https://gobyexample.com/testing and https://blog.golang.org/subtests